### PR TITLE
cleanup/fixes for plot_cmd_with_fits.py

### PR DESCRIPTION
**Pull request by lea-hagen**
Cleaned up the code a bit.  Also fixed a problem with the `savefig` keyword (similar to #603) and changed how argparse does the `log_param` keyword.

### CODE REVIEW

**Comment by karllark** [comment]
Was the `savefig` changed?  It seems to not have been.  Does it need to be?

**Comment by lea-hagen** [comment]
Yes, `savefig` is still there, but now it comes from `initialize_parser` in `beast.plotting.beastplotlib`, rather than being defined (incorrectly!) in this code.

File: beast/plotting/plot_cmd_with_fits.py [hunk 1]
```diff
@@ -205,26 +203,22 @@ def plot_cmd_with_fits(
...
-        action="store_true",
-        help="choose whether to take the log of `param` for assigning color",
-    )
-    parser.add_argument(
-        "--savefig",
-        action="store_true",
-        help="choose whether to save the figure, or show it",
+        type=int,
+        default=0,
+        help="If you would like to take the log of `param` for assigning color, set this to 1 (True)",
```
Comment (karllark) [inline beast/plotting/plot_cmd_with_fits.py:208]: Why change this to an int when it is behaving as as a boolean?

Comment (karllark) [reply beast/plotting/plot_cmd_with_fits.py:208]: Having the default be False should be action of log_param is not set with store_true, ,so not sure why this change is even needed.

**Review by karllark — CHANGES_REQUESTED** [review]

File: beast/plotting/plot_cmd_with_fits.py (hunk 1 above)
Comment (lea-hagen) [reply beast/plotting/plot_cmd_with_fits.py:208]: There were some other places we'd changed true/false keywords to ints.  But maybe that was where turning on the keyword made the input False, which was simpler to do as an int (and then changed to boolean).  Happy to change this one back.

Comment (karllark) [reply beast/plotting/plot_cmd_with_fits.py]: Great.  We should use this argparse capability where we can - less code and easier to understand what is happening including the argparse commandline help.

**Review by karllark — APPROVED** [review]
Looks good to me.

**Pull request merged**